common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1128,14 +1128,36 @@ def get_parent_location(self, locator, **kwargs):
 
         :param locator: BlockUsageLocator restricting search scope
         """
+        def is_root_block(xblock, is_root=False):
+            """ Check recursively if an xblock has a path to the course root """
+            xblock_parents = self._get_parents_from_structure(xblock, course.structure)
```

Comment:
we might be able to avoid confusion down the line if we pass `course` as an argument to this function